src/command_modules/azure-cli-webapp/azure/cli/command_modules/webapp/_params.py

```diff
@@ -2,22 +2,69 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License. See License.txt in the project root for license information.
 #---------------------------------------------------------------------------------------------
-
 from azure.cli.core.commands import register_cli_argument
-from azure.cli.core.commands.parameters import tags_type
 
 from azure.mgmt.web import WebSiteManagementClient
 from azure.cli.core.commands.client_factory import get_mgmt_service_client
-from azure.cli.core.commands.template_create import register_folded_cli_argument
+from azure.cli.core.commands.parameters import (resource_group_name_type, location_type,
+                                                CliArgumentType)
 
 # FACTORIES
-
-def _web_client_factory(**_):
+def web_client_factory(**_):
     return get_mgmt_service_client(WebSiteManagementClient)
 
+def list_webapp_names(prefix, action, parsed_args, **kwargs):#pylint: disable=unused-argument
+    if parsed_args.resource_group_name:
+        client = web_client_factory()
+        #pylint: disable=no-member
+        result = client.sites.get_sites(parsed_args.resource_group_name).value
+        return [x.name for x in result]
+
+def list_app_service_plan_names(prefix, action, parsed_args, **kwargs):#pylint: disable=unused-argument
+    if parsed_args.resource_group_name:
+        client = web_client_factory()
+        #pylint: disable=no-member
+        result = client.server_farms.get_server_farms(parsed_args.resource_group_name).value
+        return [x.name for x in result]
+
+#pylint: disable=line-too-long
 # PARAMETER REGISTRATION
+name_arg_type = CliArgumentType(options_list=('--name', '-n'), metavar='NAME')
+existing_plan_name = CliArgumentType(overrides=name_arg_type, help='The name of the app service plan', completer=list_app_service_plan_names, id_part='name')
+existing_webapp_name = CliArgumentType(overrides=name_arg_type, help='The name of the web app', completer=list_webapp_names)
+
+register_cli_argument('webapp', 'resource_group', arg_type=resource_group_name_type)
+register_cli_argument('webapp', 'location', arg_type=location_type)
+register_cli_argument('webapp', 'slot', help="the name of the slot. Default to 'productions'")
```

Comment:
i think i should remove the quot, otherwise people will think it is a valid slot name